gpkit/nomials/nomial_math.py

```diff
@@ -715,8 +718,10 @@ def __init__(self, left, right):
     def as_posyslt1(self):
         "Returns the posys <= 1 representation of this constraint."
         # todo deal with substitutions
-        raise TypeError("SignomialEquality could not simplify to"
-                        " a PosynomialInequality")
+        raise InvalidGPConstraint("SignomialEquality could not simplify"
+                                  " to a PosynomialInequality; try forming"
+                                  " your program as SignomialProgram or"
+                                  " calling .localsolve().")
```

Comment:
yes; what I meant was, what user(s) will encounter this error message, and will it be clear to them what to do next? (e.g. note that `SignomialProgram` is not designed to be user facing)